### ISSUE TITLE
Bot-boosts from a Home-excluded List still appear in Home timeline

### ISSUE THREAD

**Issue by SmartmanApps** [mastodon/mastodon#28331]
### Steps to reproduce the problem

1. Follow a bot that boosts Posts (such as @bot@dotnet.social)
2. Add it to a List
3. Exclude that List's Posts from appearing in the Home timeline

### Expected behaviour

Posts should ONLY appear in the List timeline

### Actual behaviour

Bot-boosted Posts are still appearing in the Home timeline

### Detailed description

Here is a screenshot of the bug in action. My Tech List is excluded from appearing in Home timeline. A Post boosted by the dotnet bot appears in my Tech list, and is further identified as having been boosted by the bot. Same post appears in my Home timeline - it shouldn't (I don't follow that person, and am only seeing the post because the dotnet bot boosted it) - and further to that is NOT identified as having been boosted by the bot. This is happenning with every Post boosted by the dotnet bot. Somewhere there is a bug which is causing the system to NOT check if a Post from a bot is in a Home-excluded List and posts it in both timelines.
![image](https://github.com/mastodon/mastodon/assets/20635275/f178f6a1-27ad-4b0f-a543-72a6b8099f4b)

### Mastodon instance

dotnet.social

### Mastodon version

v4.2.3

### Browser name and version

Chrome 120.0.6099.71

### Operating system

Windows 22H2 19045.3693

**Comment by SmartmanApps** [comment]
P.S. seems to be not JUST bot-boosted posts. I have myself in the same Home-excluded "Tech" List, and I just boosted something, and it also appeared in my Home timeline. I have previously seen this issue [github.com/mastodon/mastodon/issues/26746](https://github.com/mastodon/mastodon/issues/26746) and so now I'm not sure if this is the same issue or just a related issue. In any case  (bot-boosted or self-boosted or self-posted) we have content from a Home-excluded List leaking into the Home timeline.

**Comment by ClearlyClaire** [comment]
Do you perhaps follow the #GameDev or #DotNet hashtag?

**Comment by SmartmanApps** [comment]
> Do you perhaps follow the #GameDev or #DotNet hashtag?

Actually yes I was without realising it! Not sure how that happened - I must've done that before I found out about the dotnet bot. I've unfollowed the hashtags now and will confirm this solves the issue before closing this issue. In regards to the other issue I mentioned though, that doesn't explain why my own content is still appearing in my Home timeline (but that's a separate issue). Thanks!
